google/cloud/bigtable/tests/instance_admin_emulator.cc

```diff
@@ -187,17 +195,21 @@ class InstanceAdminEmulator final
   grpc::Status ListClusters(grpc::ServerContext* context,
                             btadmin::ListClustersRequest const* request,
                             btadmin::ListClustersResponse* response) override {
-    std::string prefix = request->parent();
-
-    // magical instance name "-" must return all clusters in the project
-    if (request->parent().back() == '-') {
-      for (auto const& kv : clusters_) {
-        *response->add_clusters() = kv.second;
+    for (auto const& instance : instances_) {
+      if (std::string::npos == instance.first.find(request->parent().substr(
```

Comment:
This is a little obscure, and not very efficient.  I think you are trying to get the project name from the `request->parent()`, do you mind hoisting that outside the loop and giving it a variable name?  Like maybe:

```C++
auto project_name = request->parent().substr(0, request->parent().find_last_of('/'));
for (auto const& instance : instances_) {
...
```